### PR TITLE
Chore: Don't cache static data tables (ui/elements, ui/data, kb layouts & co)

### PR DESCRIPTION
**Pull request by NiLuJe**
Using `require` didn't make much sense since most of them are only used in a single place anyway, and it takes care of a few weird interactions in the process (besides not polluting `package.loaded` with useless crap ;)).

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12519)

### CODE REVIEW

**Comment by hius07** [comment]
Huh, cool.

**Comment by Commodore64user** [comment]
ohhh, this is what that other comment meant. I guess... 

but wouldn't this also need adjusting then?

https://github.com/koreader/koreader/blob/bf290e0c2b6262c14b57e922a89011c2402102d8/frontend/ui/menusorter.lua#L27-L29

**Comment by NiLuJe** [comment]
That one slipped through my regex as it's user data, but, yeah ;).

**Comment by hius07** [comment]
> ohhh, this is what that other comment meant. I guess...

Exactly, thanks. And wow.

**Comment by Commodore64user** [comment]
> Exactly, thanks. And wow.

I beg your pardon?

File: frontend/ui/menusorter.lua [hunk 1]
```diff
@@ -22,11 +21,10 @@ local MenuSorter = {
 function MenuSorter:readMSSettings(config_prefix)
     if config_prefix then
         local menu_order = string.format(
-            "%s/%s_menu_order", DataStorage:getSettingsDir(), config_prefix)
+            "%s/%s_menu_order.lua", DataStorage:getSettingsDir(), config_prefix)
 
-        if lfs.attributes(menu_order..".lua") then
-            return require(menu_order) or {}
-        end
+        local ok, data = pcall(dofile, menu_order)
```
Comment (Frenzie) [inline frontend/ui/menusorter.lua:26]: I prefer the original (not talking about the require). Okay, so the user made a typo… but that's the point! They should be immediately notified that they messed up, shouldn't they?

Comment (Frenzie) [reply frontend/ui/menusorter.lua:26]: Concretely, this is conflating "file exists" and "file has syntax error".

I think you can alternatively write it along these lines:
```lua
if not ok and not data:match(menu_order) then
    -- Syntax error, it'll contain the filename in a file not found error
    -- `cannot open bla.lua: No such file or directory`
    error("immediate proper feedback")
end

**Comment by Frenzie** [comment]
> Using require didn't make much sense since most of them are only used in a single place anyway

I can't seem to find the relevant page atm, but I think loadstring is only ever interpreted. Is dofile compiled into bytecode?

**Comment by NiLuJe** [comment]
> I can't seem to find the relevant page atm, but I think loadstring is only ever interpreted. Is dofile compiled into bytecode?

`dofile` is usually implemented as a wrapper around `loadfile`.

c.f., https://github.com/LuaJIT/LuaJIT/blob/87ae18af97fd4de790bb6c476b212e047689cc93/src/luajit.c#L168-L172

And https://www.lua.org/pil/8.html for more context perhaps?

They all compile to bytecde, AFAICT, c.f., the description of `load`: http://luajit.org/extensions.html

**Comment by Frenzie** [comment]
I was referring to the Not Yet Implemented page, which seems to have vanished (and as such this copy is therefore outdated): https://web.archive.org/web/20220708225331/http://wiki.luajit.org/NYI

Anyway, I probably misremembered. Whether the *call* is compiled is irrelevant.

**Comment by NiLuJe** [comment]
Oh, yeah, no clue where that went ^^.

But on that note, fun fact: *coroutines* are NYI, so they're not JIT'ed ;).

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 37 of 37 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/12519)* status: all files reviewed, 1 unresolved discussion (waiting on @Frenzie)

**Pull request merged**